lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,878 @@
+# ansible all -m postgresql_facts -a 'filter=dat*,rol*'
+
+# Display only replications and repl_slots facts from standby hosts using shell-style wildcards:
+# ansible standby -m postgresql_facts -a 'filter=repl*'
+
+# Display all facts from databases hosts except settings:
+# ansible databases -m postgresql_facts -a 'filter=!settings'
+
+- name: Collect PostgreSQL version and extensions
+  postgresql_facts:
+    filter: ver*,ext*
+
+- name: Collect all facts except settings and roles
+  postgresql_facts:
+    filter: "!settings,!roles"
+
+- name: Collect tablespaces and repl_slots facts
+  postgresql_facts:
+    filter:
+    - tablesp*
+    - repl_sl*
+
+- name: Collect all facts except settings
+  postgresql_facts:
+    filter:
+    - "!settings"
+'''
+
+RETURN = '''
+ansible_facts:
+  description: Dictionary containing the detailed information about the PostgreSQL instance.
+  returned: always
+  type: complex
+  contains:
+    version:
+      description: Database server version U(https://www.postgresql.org/support/versioning/).
+      returned: always
+      type: dict
+      contains:
+        major:
+          description: Major server version.
+          returned: always
+          type: int
+          sample: 11
+        minor:
+          description: Minor server version.
+          returned: always
+          type: int
+          sample: 1
+    databases:
+      description: Information about databases.
+      returned: always
+      type: dict
+      contains:
+        database_name:
+          description: Database name.
+          returned: always
+          type: dict
+          sample: template0
+          contains:
+            access_priv:
+              description: Database access privileges.
+              returned: always
+              type: str
+              sample: "=c/postgres_npostgres=CTc/postgres"
+            collate:
+              description:
+              - Database collation U(https://www.postgresql.org/docs/current/collation.html).
+              returned: always
+              type: str
+              sample: en_US.UTF-8
+            ctype:
+              description:
+              - Database LC_CTYPE U(https://www.postgresql.org/docs/current/multibyte.html).
+              returned: always
+              type: str
+              sample: en_US.UTF-8
+            encoding:
+              description:
+              - Database encoding U(https://www.postgresql.org/docs/current/multibyte.html).
+              returned: always
+              type: str
+              sample: UTF8
+            owner:
+              description:
+              - Database owner
+                U(https://www.postgresql.org/docs/current/sql-createdatabase.html).
+              returned: always
+              type: str
+              sample: postgres
+            size:
+              description: Database size.
+              returned: always
+              type: str
+              sample: 7861 kB
```

Comment:
This ought to be in bytes as well. So that these values can be used without complexity (i.e. being parsed).

Reply:
Changed  to bytes